packages/@webex/plugin-cc/src/AgentProfileService/AgentProfileService.ts

```diff
@@ -0,0 +1,146 @@
+import {WebexSDK, HTTP_METHODS} from '../types';
+import {
+  DesktopProfileResponse,
+  ListAuxCodesResponse,
+  ListTeamsResponse,
+  UserResponse,
+} from './types';
+
+export default class AgentProfileService {
+  ciUserId: string;
+  orgId: string;
+  webex: WebexSDK;
+  wccAPIURL: string;
+
+  constructor(ciUserId: string, orgId: string, webex: WebexSDK, wccAPIURL: string) {
+    this.ciUserId = ciUserId;
+    this.orgId = orgId;
+    this.webex = webex;
+    this.wccAPIURL = wccAPIURL;
+  }
+
+  /**
+   * Method to get User using CI.
+   * @param {string} ciUserId The CI Id of the user.
+   * @param {string} orgId The ID of an Organization.
+   * @returns {Promise<UserResponse>} A promise that eventually resolves to an API response.
+   */
+
+  public async getUserUsingCI(ciUserId: string, orgId: string): Promise<UserResponse> {
+    try {
+      if (!ciUserId || !orgId) Promise.reject(new Error('ciUserId or orgId is undefined'));
+
+      const URL = `${this.wccAPIURL}organization/${orgId}/user/by-ci-user-id/${ciUserId}`;
+      this.webex.logger.log('getUserUsingCI api called successfully.');
```

Comment:
Isn't this incorrect? We haven't called the get request yet. It's being called in line 36. So why are we logging this here..

Reply:
Good Catch. We should print logger message after the API call is successful.